stl/inc/expected

```diff
@@ -722,6 +1023,24 @@ public:
     }
 
 private:
+    // These overloads force copy elision from the invoke call into _Value
+    template <class _Fn, class _Ux>
+    constexpr expected(_Construct_expected_from_invoke_result_tag, _Fn&& _Func, _Ux&& _Arg) noexcept(
+        is_nothrow_constructible_v<_Ty, invoke_result_t<_Fn, _Ux>>)
+        : _Value(_STD invoke(_STD forward<_Fn>(_Func), _STD forward<_Ux>(_Arg))), _Has_value{true} {}
+
+    // For when transform is called on an expected<void,E>

```

Comment:
We have a different partial specialization for cv-`void` value types; I suspect this constructor should be relocated. (Suggests lack of test coverage.)

Reply:
Yeah, but this is for when a user calls `expected<void,E>transform(F)`, thus constructing a `expected<invoke_result_t<F>,E>`, requiring a constructor for `expected<T,E>` which calls the `F` instance with no argument.

Reply:
Don't know how I managed to misunderstand that despite the comment. 🤷